src/lib/lessons.ts

```diff
@@ -49,38 +75,97 @@ async function loadLessonMetadataFromSanity(slug: string) {
+
+    return {}
+  }
+}
+
+const deriveDataFromBaseValues = (result: any) => {
+  const http_url = `${process.env.NEXT_PUBLIC_DEPLOY_URL}${result.path}`
+  const lesson_view_url = `${process.env.NEXT_PUBLIC_AUTH_DOMAIN}/api/v1${result.path}/views`
+  const download_url = `${process.env.NEXT_PUBLIC_AUTH_DOMAIN}/api/v1${result.path}/signed_download`
+
+  return {http_url, lesson_view_url, download_url}
+}
+
+async function loadLessonMetadataFromGraphQL(slug: string, token?: string) {
+  const graphQLClient = getGraphQLClient(token)
+
+  try {
+    const {lesson: lessonMetadataFromGraphQL} = await graphQLClient.request(
+      loadLessonGraphQLQuery,
+      {slug},
+    )
+
+    return lessonMetadataFromGraphQL
+  } catch (e) {
+    // Likely a 404 Not Found error
+    console.log('Error fetching from GraphQL: ', e)
 
-  return lesson
+    return {}
+  }
 }
 
 export async function loadLesson(
   slug: string,
   token?: string,
 ): Promise<LessonResource> {
   token = token || getAccessTokenFromCookie()
-  const graphQLClient = getGraphQLClient(token)
-
-  const variables = {
-    slug: slug,
-  }
 
-  const {lesson} = await graphQLClient.request(
-    loadLessonGraphQLQuery,
-    variables,
+  /******************************************
+   * Primary Lesson Metadata GraphQL Request
+   * ****************************************/
+  const lessonMetadataFromGraphQL = await loadLessonMetadataFromGraphQL(
+    slug,
+    token,
   )
 
-  const lessonMetadataFromSanity = await loadLessonMetadataFromSanity(slug)
-
-  const lessonMetadata = {...lesson, ...lessonMetadataFromSanity}
-
+  /**********************************************
+   * Load comments from separate GraphQL Request
+   * ********************************************/
   // comments are user-generated content that must come from the egghead-rails
   // backend, so load those separately from the rest of lesson metadata.
   const comments = await loadLessonComments(slug, token)
 
+  /*************************************
+   * Sanity Request for Lesson Metadata
+   * ***********************************/
+  // this will be used to override values from graphql
+  const lessonMetadataFromSanity = await loadLessonMetadataFromSanity(slug)
+
+  /*************************************
+   * Merge All Lesson Metadata Together
+   * ***********************************/
+  // with preference for data coming from Sanity
+  const lessonMetadata = mergeLessonMetadata(
+    lessonMetadataFromGraphQL,
+    lessonMetadataFromSanity,
+  )
+
+  // if we aren't able to find Lesson metadata at either source, throw an
+  // error.
+  if (isEmpty(lessonMetadata.slug)) {
+    throw new Error(`Unable to lookup lesson metadata (slug: ${slug})`)
+  }
+
   return {...lessonMetadata, comments} as LessonResource
 }
 
+// values in the graphql that are being skipped/ignored by Sanity
+//
+// - dash_url - not used
+// - staff_notes_url - not used
+// - icon_url - this is being used by the lesson page
```

Comment:
The `icon_url`, when coming from egghead-rails, is an image attached to the lesson. If one of the lesson tags has a thumb-size icon, we use that. Otherwise, there is a fallback egghead svg.
https://github.com/eggheadio/egghead-rails/blob/d8a9480088e3196c9844250218d4fc146c7b5fcd/app/models/concerns/content_model.rb#L89-L100

We could approximate the same thing client-side by grabbing the tag images and/or using the fallback. Then Sanity doesn't need to know about it. Unless, we want to start doing custom `icon_url` values for lessons.